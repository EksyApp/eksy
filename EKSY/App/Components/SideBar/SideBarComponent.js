import React, {Component} from 'react'
import {View, Text} from 'react-native'
import SideBarLogo from '../../Components/SideBar/SideBarLogo'
import {Button} from '../../Components/Common/index'
import * as Theme from '../../Theme/index'
import PropTypes from 'prop-types'
import {UserShape} from "../../Utils/PropTypeShapes";

//Renders the sidebar view that holds buttons to navigate in app
//Go to map closes the sidebar and returns to MainViewComponent
//Settings takes to SettingsComponent
//Add marker takes to AddMarkerComponent
//Login takes to LoginComponent or SignUpComponent
//Admin takes to AdminToolsComponent
export default class SideBarComponent extends Component {

	renderLoginButton() {
		if (!this.props.user) {
			return (
					<Button onPress={this.props.goToLoginScreen}>
						Login
					</Button>
			)
		}
	}

	renderAddMarkerButton() {
		if (this.props.user) {
			return (
					<Button onPress={this.props.goToAddMarker}>
						Add a marker
					</Button>
			)
		}
	}

	renderAdminButton() {
		if (this.props.user && this.props.user.admin) {
			return (
					<Button onPress={this.props.goToAdminTools}>
						Admin
					</Button>
			)
		}
	}

	render() {
		return (
				<View style={styles.menubarStyle}>
					<View style={styles.logoStyle}>
						<SideBarLogo />
					</View>
					<View style={styles.buttonList}>
						<Button onPress={this.props.goToMap}>
							Go to map
						</Button>
						<Button onPress={this.props.goToUserSettings}>
							Settings
						</Button>
						{this.renderAddMarkerButton()}
						{this.renderLoginButton()}
						{this.renderAdminButton()}
					</View>
					{__DEV__ && <Text style={{flex: 1, marginLeft: 20, fontSize: 12}}>debug version</Text>}
				</View>
		)
	}
}

const styles = {
	menubarStyle: {
		flex: 1,
		flexDirection: 'column',
		// Placeholder background color
		backgroundColor: Theme.backgroundColor,
	},
	buttonList: {
		flexDirection: 'column',
		justifyContent: 'space-around',
	},
	logoStyle: {
		// Three blue lines are from the logo picture
	}
}

SideBarComponent.propTypes = {
	user: UserShape,
	goToLoginScreen: PropTypes.func,
	goToAddMarker: PropTypes.func,
	goToAdminTools: PropTypes.func,
	goToMap: PropTypes.func,
	goToUserSettings: PropTypes.func,
}
