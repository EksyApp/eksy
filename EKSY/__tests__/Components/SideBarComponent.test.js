import 'react-native';
import React from 'react';
import renderer from 'react-test-renderer';
import SideBarComponent from "../../App/Components/SideBar/SideBarComponent";

jest.mock('react-native-fetch-blob', () => {
	return {
		DocumentDir: () => {},
		polyfill: () => {}
	}
})

test('SideBarComponent renders correctly', () => {
	const tree = renderer.create(
			<SideBarComponent />
	).toJSON();
	expect(tree).toMatchSnapshot();
});