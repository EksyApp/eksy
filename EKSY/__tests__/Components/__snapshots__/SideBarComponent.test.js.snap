// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SideBarComponent renders correctly 1`] = `
<View
  style={
    Object {
      "backgroundColor": "rgb(243,244,229)",
      "flex": 1,
      "flexDirection": "column",
    }
  }
>
  <View
    style={Object {}}
  >
    <View
      onLayout={[Function]}
      style={
        Array [
          undefined,
          Object {
            "marginBottom": 20,
            "width": "100%",
          },
          Object {
            "height": 54,
            "width": "100%",
          },
        ]
      }
    >
      <Image
        accessibilityComponentType={undefined}
        accessibilityLabel={undefined}
        accessibilityTraits={undefined}
        accessible={true}
        hitSlop={undefined}
        onLayout={undefined}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        resizeMode="cover"
        source={1}
        style={
          Array [
            Object {},
            Object {
              "height": 54,
              "width": "100%",
            },
          ]
        }
        testID={undefined}
      />
    </View>
  </View>
  <View
    style={
      Object {
        "flexDirection": "column",
        "justifyContent": "space-around",
      }
    }
  >
    <View
      accessibilityComponentType={undefined}
      accessibilityLabel={undefined}
      accessibilityTraits={undefined}
      accessible={true}
      hitSlop={undefined}
      isTVSelectable={true}
      onLayout={undefined}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "alignSelf": "stretch",
          "backgroundColor": "rgb(230,232,227)",
          "borderColor": "rgb(101,114,122)",
          "borderRadius": 5,
          "borderWidth": 1,
          "height": 40,
          "marginBottom": 15,
          "marginLeft": 5,
          "marginRight": 5,
          "marginTop": 5,
          "opacity": 1,
        }
      }
      testID={undefined}
      tvParallaxProperties={undefined}
    >
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Object {
            "alignSelf": "center",
            "color": "rgb(101,114,122)",
            "fontSize": 16,
            "fontWeight": "600",
            "paddingBottom": 10,
            "paddingTop": 10,
          }
        }
      >
        Go to map
      </Text>
    </View>
    <View
      accessibilityComponentType={undefined}
      accessibilityLabel={undefined}
      accessibilityTraits={undefined}
      accessible={true}
      hitSlop={undefined}
      isTVSelectable={true}
      onLayout={undefined}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "alignSelf": "stretch",
          "backgroundColor": "rgb(230,232,227)",
          "borderColor": "rgb(101,114,122)",
          "borderRadius": 5,
          "borderWidth": 1,
          "height": 40,
          "marginBottom": 15,
          "marginLeft": 5,
          "marginRight": 5,
          "marginTop": 5,
          "opacity": 1,
        }
      }
      testID={undefined}
      tvParallaxProperties={undefined}
    >
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Object {
            "alignSelf": "center",
            "color": "rgb(101,114,122)",
            "fontSize": 16,
            "fontWeight": "600",
            "paddingBottom": 10,
            "paddingTop": 10,
          }
        }
      >
        Settings
      </Text>
    </View>
    <View
      accessibilityComponentType={undefined}
      accessibilityLabel={undefined}
      accessibilityTraits={undefined}
      accessible={true}
      hitSlop={undefined}
      isTVSelectable={true}
      onLayout={undefined}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "alignSelf": "stretch",
          "backgroundColor": "rgb(230,232,227)",
          "borderColor": "rgb(101,114,122)",
          "borderRadius": 5,
          "borderWidth": 1,
          "height": 40,
          "marginBottom": 15,
          "marginLeft": 5,
          "marginRight": 5,
          "marginTop": 5,
          "opacity": 1,
        }
      }
      testID={undefined}
      tvParallaxProperties={undefined}
    >
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Object {
            "alignSelf": "center",
            "color": "rgb(101,114,122)",
            "fontSize": 16,
            "fontWeight": "600",
            "paddingBottom": 10,
            "paddingTop": 10,
          }
        }
      >
        Login
      </Text>
    </View>
  </View>
</View>
`;
